OPHD/States/CrimeRateUpdate.cpp

```diff
@@ -0,0 +1,88 @@
+#include "CrimeRateUpdate.h"
+#include "../UI/PopulationPanel.h"
+#include "../Things/Structures/Structure.h"
+#include "../StructureManager.h"
+#include <NAS2D/Utility.h>
+
+namespace CrimeRate
+{
+	bool isProtectedByPolice(const std::vector<TileList>& policeOverlays, Structure* structure);
+	int calculateMoraleChange(int meanCrimeRate);
+	void setPopulationPanel(int moraleChange, int meanCrimeRate, PopulationPanel& populationPanel);
+
+
+	int update(const std::vector<TileList>& policeOverlays, PopulationPanel& populationPanel)
+	{
+		const auto& structuresWithCrime = NAS2D::Utility<StructureManager>::get().structuresWithCrime();
+
+		// Colony will not have a crime rate until at least one structure that supports crime is built
+		if (structuresWithCrime.empty()) {
+			setPopulationPanel(0, 0, populationPanel);
+			return 0;
+		}
+
+		double accumulatedCrime = 0;
+
+		for (auto structure : structuresWithCrime)
```

Comment:
If you combine this loop with the loop in the function `isProtectedByPolice`, you end up with an O(n^2) algorithm. I think we could tamp it down to O(n) by reformatting `policeOverlays` into a graph that the values could be pulled off of in an O(1) fashion. Of course, then you are paying for the creation of the graph. 

Another solution could be to try some sort of binary search here.

I am OK addressing now or leaving and seeing if it becomes a performance issue, but wanted to point it out for specific review. Someone else may have a better solution on how to address than what I have suggested.